### PR TITLE
Refactor env variables 

### LINKED ISSUE #60: Enhance : Look for env variables : Database connection if not fallback to config
Plan is to look for the env variables if not fall back to the config for raw_data section off the API , since backend is now part of the codebase both can read from .env

### LINKED ISSUE #62: [BUG] Fix Failing Unit tests and Build
With the change in request param on /snapshot build and unit tests are failing

### PR DESCRIPTION
**Pull request by kshitijrajsharma**
- Resolve #60 
- Resolve #62
- Disable /latest/ in versions  
- Improve unit tests

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -38,28 +39,110 @@
...
+    logging.error(
+        "value not supported for file_upload_method ,switching to default disk method"
+    )
+    USE_S3_TO_UPLOAD = False
+
+if FILE_UPLOAD_METHOD == "s3":
+    USE_S3_TO_UPLOAD = True
+    BUCKET_NAME = config.get(
+        "EXPORT_UPLOAD", "BUCKET_NAME", fallback=os.environ.get("BUCKET_NAME")
+    )
```
Comment (eternaltyro) [inline src/config.py:119]: os.environ.get(something, config(section, key, fallback=??))

File: src/config.py [hunk 2]
```diff
@@ -75,66 +158,94 @@
 logging.getLogger("boto").propagate = False  # disable boto3 logging
 
 
-logger = logging.getLogger("export_tool_api")
+logger = logging.getLogger("raw_data_api")
+
+EXPORT_PATH = config.get(
+    "API_CONFIG", "EXPORT_PATH", fallback=os.environ.get("EXPORT_PATH", "exports")
+)
```
Comment (eternaltyro) [inline src/config.py:165]: env lookup before config lookup

File: src/config.py [hunk 3]
```diff
@@ -75,66 +158,94 @@
...
-    os.makedirs(export_path)
-allow_bind_zip_filter = config.get("API_CONFIG", "allow_bind_zip_filter", fallback=None)
-if allow_bind_zip_filter:
-    allow_bind_zip_filter = True if allow_bind_zip_filter.lower() == "true" else False
+    os.makedirs(EXPORT_PATH)
+ALLOW_BIND_ZIP_FILTER = config.get(
+    "API_CONFIG",
+    "ALLOW_BIND_ZIP_FILTER",
+    fallback=os.environ.get("ALLOW_BIND_ZIP_FILTER", None),
+)
```
Comment (eternaltyro) [inline src/config.py:174]: env lookup should take precedence over config lookup

File: src/config.py [hunk 4]
```diff
@@ -75,66 +158,94 @@
...
+
 
-AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, BUCKET_NAME = None, None, None
 # check either to use connection pooling or not
-use_connection_pooling = config.getboolean(
-    "API_CONFIG", "use_connection_pooling", fallback=False
+USE_CONNECTION_POOLING = config.getboolean(
+    "API_CONFIG",
+    "USE_CONNECTION_POOLING",
+    fallback=os.environ.get("USE_CONNECTION_POOLING", False),
```
Comment (eternaltyro) [inline src/config.py:183]: precedence: env over config

File: src/config.py [hunk 5]
```diff
@@ -75,66 +158,94 @@
...
+            "host": config.get("DB", "PGHOST", fallback=os.environ.get("PGHOST")),
+            "port": config.get("DB", "PGPORT", fallback=os.environ.get("PGPORT")),
+            "dbname": config.get(
+                "DB", "PGDATABASE", fallback=os.environ.get("PGDATABASE")
+            ),
+            "user": config.get("DB", "PGUSER", fallback=os.environ.get("PGUSER")),
+            "password": config.get(
+                "DB", "PGPASSWORD", fallback=os.environ.get("PGPASSWORD")
+            ),
+        }
```
Comment (eternaltyro) [inline src/config.py:207]: Precedence: env should be higher than config

File: src/config.py [hunk 6]
```diff
@@ -75,66 +158,94 @@
...
+    )
+    secret_key = config.get(
+        "OAUTH", "APP_SECRET_KEY", fallback=os.environ.get("APP_SECRET_KEY")
+    )
+    login_redirect_uri = config.get(
+        "OAUTH", "LOGIN_REDIRECT_URI", fallback=os.environ.get("LOGIN_REDIRECT_URI")
+    )
+    scope = config.get(
+        "OAUTH", "OSM_PERMISSION_SCOPE", fallback=os.environ.get("OSM_PERMISSION_SCOPE")
+    )
```
Comment (eternaltyro) [inline src/config.py:239]: Precedence check

**Review by eternaltyro — CHANGES_REQUESTED** [review]
Please check precedence of acquiring values for parameters.

File: docs/src/installation/configurations.md [hunk 7]
```diff
@@ -81,29 +82,28 @@ Considering You have PSQL-POSTGIS setup with user **postgres** host **localhost*
...
 [API_CONFIG]
-env=dev
-log_level=debug
-limiter_storage_uri=redis://redis:6379
+LOG_LEVEL=debug
+RATE_LIMITER_STORAGE_URI=redis://redis:6379
 
-[CELERY]
-CELERY_BROKER_URL=redis://redis:6379/0
-CELERY_RESULT_BACKEND=redis://redis:6379/0
```
[the remark this answers is not in the document]
Comment (kshitijrajsharma) [reply docs/src/installation/configurations.md:106]: Its on the doc ! on configuration.md also on docker.md

**Pull request merged**